test/kubernetes/integration_test.go

```diff
@@ -0,0 +1,208 @@
+	if err := teardown(); err != nil {
+		// don't call log.Fatalln() as we exit with `m.Run()`'s exit status
+		log.Println(err)
+	}
+
+	os.Exit(exitStatus)
+}
+
+func TestPod_Single_Volume(t *testing.T) {
+	volumeName := "my-do-volume"
+	claimName := "csi-pod-pvc"
+
+	pod := &v1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "my-csi-app",
+		},
+		Spec: v1.PodSpec{
+			Containers: []v1.Container{
+				{
+					Name:  "my-busybox",
+					Image: "busybox",
+					VolumeMounts: []v1.VolumeMount{
+						{
+							MountPath: "/data",
+							Name:      volumeName,
+						},
+					},
+					Command: []string{
+						"sleep",
+						"1000000",
+					},
+				},
+			},
+			Volumes: []v1.Volume{
+				{
+					Name: volumeName,
+					VolumeSource: v1.VolumeSource{
+						PersistentVolumeClaim: &v1.PersistentVolumeClaimVolumeSource{
+							ClaimName: claimName,
+						},
+					},
+				},
+			},
+		},
+	}
+
+	fmt.Println("Creating pod")
+	_, err := client.CoreV1().Pods(namespace).Create(pod)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	pvc := &v1.PersistentVolumeClaim{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: claimName,
+		},
+		Spec: v1.PersistentVolumeClaimSpec{
+			AccessModes: []v1.PersistentVolumeAccessMode{
+				v1.ReadWriteOnce,
+			},
+			Resources: v1.ResourceRequirements{
+				Requests: v1.ResourceList{
+					v1.ResourceStorage: resource.MustParse("5Gi"),
+				},
+			},
+			StorageClassName: strPtr("do-block-storage"),
+		},
+	}
+
+	fmt.Println("Creating pvc")
+	_, err = client.CoreV1().PersistentVolumeClaims(namespace).Create(pvc)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	fmt.Println("Waiting pod to be running ...")
+	ctx, cancel := context.WithTimeout(context.Background(), time.Minute*5)
+	defer cancel()
+
+	waitForPod(ctx, client, pod.Name)
+
+	fmt.Println("Finished!")
+}
+
+func setup() error {
+	var kubeconfig string
+	flag.StringVar(&kubeconfig, "kubeconfig", "", "absolute path to the kubeconfig file")
+	flag.Parse()
+
+	if kubeconfig == "" {
+		kubeconfig = os.Getenv(clientcmd.RecommendedConfigPathEnvVar)
+		fmt.Println("==> Using kubeconfig from env(KUBECONFIG) path:", kubeconfig)
+	}
+
+	if kubeconfig == "" {
+		return errors.New("neither --kubeconfig nor KUBECONFIG is specified")
+	}
+
+	// use the current context in kubeconfig
+	config, err := clientcmd.BuildConfigFromFlags("", kubeconfig)
+	if err != nil {
+		return err
+	}
+
+	// create the clientset
+	client, err = kubernetes.NewForConfig(config)
+	if err != nil {
+		return err
+	}
+
+	// create test namespace
+	_, err = client.CoreV1().Namespaces().Create(&v1.Namespace{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: namespace,
+		},
+	})
+	if err != nil && !kubeerrors.IsAlreadyExists(err) {
+		return err
+	}
+
+	return nil
+}
+
+func teardown() error {
+	// delete all test resources
+	err := client.CoreV1().Namespaces().Delete(namespace, nil)
+	if err != nil && !(kubeerrors.IsNotFound(err) || kubeerrors.IsAlreadyExists(err)) {
+		return err
+	}
+
```

Comment:
💭 do we need to wait for complete namespace deletion?
rationale: if another test tries to reuse a namespace it will fail as the namespace is still being deleted

Reply:
@lxfontes yes we need, if you run consecutive runs, it'll fail (I've seen it several times). I'm actually aware of this, this check is only temporary, I wanted to add more bullet prof checks later in the process. I'll make sure to fix it indeed 👍